### ISSUE TITLE
[EIP-6963] Multi Injected Provider

### ISSUE THREAD

**Issue by anxolin** [cowprotocol/cowswap#3987]

**Comment by anxolin** [comment]
Potentially solved by https://github.com/cowprotocol/cowswap/issues/3987

**Comment by omarabid** [comment]
@anxolin Can you provide an example of how to get the provider out of Wagmi? I tried different ways (including window.ethereum after the injectedwallet is loaded) and still couldn't get the widget to recognize the connection.

**Comment by alfetopito** [comment]
> @anxolin Can you provide an example of how to get the provider out of Wagmi? I tried different ways (including window.ethereum after the injectedwallet is loaded) and still couldn't get the widget to recognize the connection.

We are not using Wagmi on CoW Swap. Unless you mean in the CoW Swap Widget?

**Comment by omarabid** [comment]
> > @anxolin Can you provide an example of how to get the provider out of Wagmi? I tried different ways (including window.ethereum after the injectedwallet is loaded) and still couldn't get the widget to recognize the connection.
> 
> We are not using Wagmi on CoW Swap. Unless you mean in the CoW Swap Widget?

Yes. I am using the widget with Wagmi and I am wondering how to get the provider to pass to CowSwapWidget.

**Comment by shoom3301** [comment]
@omarabid hi!
You can use it like that: https://github.com/cowprotocol/cowswap/pull/3262/files#diff-9b08237b32c901eadab5f680238fb314c19270df9177ef9170ec9846aa416f60R11
